sonic-slave-buster/Dockerfile.j2

```diff
@@ -40,11 +48,88 @@ RUN echo "deb [arch=arm64] http://deb.debian.org/debian buster main contrib non-
         echo 'deb [arch=arm64] http://ftp.debian.org/debian buster-backports main' >> /etc/apt/sources.list && \
         echo "deb [arch=arm64] http://packages.trafficmanager.net/debian/debian buster main contrib non-free" >> /etc/apt/sources.list && \
         echo "deb [arch=arm64] http://packages.trafficmanager.net/debian/debian buster-updates main contrib non-free" >> /etc/apt/sources.list
+{%- elif CONFIGURED_ARCH == "armhf" and CROSS_BUILD_ENVIRON == "y" %}
```

Comment:
If i have to install a package lets say (libpopt-dev) which has both the amd64 & arm64 variants available. 

If i install it using "apt-get install libpopt-dev -y", which arch will be fetched from and how does it work? I'm asking this because from what i understand the the "/etc/apt/sources.list" file is not modified if CROSS_BUILD_ENVIRON == y.

Reply:
If you install "apt-get install libpopt-dev -y" then amd64 will be installed (as the sonic-slave image architecture which is amd64). If you install "apt-get install libpopt-dev:arm64 -y" then arm64 package will be installed. The same logic for the ':armhf' package suffix. You can see it in the sonic-slave-buster/Dockerfile.j2 file.

Reply:
Here is a sample from the Dockerfile.j2:

**RUN apt-get update && apt-get install -y libelf-dev:$arch libdw-dev:$arch  libbz2-dev:$arch liblzo2-dev:$arch libedit-dev:$arch libevent-dev:$arch libopts25-dev:$arch ...**

where $arch is either arm64 or armhf according to the target architecture.